src/silo/preprocessing/metadata_validator.cpp

```diff
@@ -0,0 +1,30 @@
+#include "silo/preprocessing/metadata_validator.h"
+#include "silo/preprocessing/preprocessing_exception.h"
+
+namespace silo::preprocessing {
+
+MetadataValidator::MetadataValidator(const config::ConfigRepository& config_repository)
+    : config_repository_(config_repository) {}
+
+void MetadataValidator::validateMedataFile(
+   const std::filesystem::path& metadata_file,
+   const std::filesystem::path& config_file
+) const {
+   const auto csv_reader = MetadataReader::getReader(metadata_file);
+   const auto config = config_repository_.getValidatedConfig(config_file);
+   const auto metadata_columns = csv_reader.get_col_names();
+
+   for (const auto& config_column : config.schema.metadata) {
```

Comment:
See Chaoran's comment in the ticket: I think we should also check the other way round and maybe write a warning to the log file that there are additional columns?